File: frameworks/Scala/play2-scala/play2-scala-anorm/app/controllers/Application.scala
package controllers

import javax.inject.{Inject, Singleton}

import play.api.mvc._
import play.mvc.Http
import play.api.libs.json.Json
import java.util.concurrent._
import models.{WorldDAO, FortunesDAO, World, Fortune}
import utils.DbOperation
import scala.concurrent.{Future, ExecutionContext}

@Singleton()
class Application @Inject() (fortunesDAO: FortunesDAO, worldDAO: WorldDAO, dbOperation: DbOperation, val controllerComponents: ControllerComponents)(implicit ec: ExecutionContext)
  extends BaseController {

  def getRandomWorlds(n: Int): Future[Seq[World]] = dbOperation.asyncDbOp { implicit connection =>
    for (_ <- 1 to n) yield {
      worldDAO.findById(getNextRandom)
    }
  }

  def getFortunes: Future[Seq[Fortune]] = dbOperation.asyncDbOp { implicit connection =>
    fortunesDAO.getAll
  }

  def updateWorlds(n: Int): Future[Seq[World]] = dbOperation.asyncDbOp { implicit connection =>
    for(_ <- 1 to n) yield {
      val world = worldDAO.findById(getNextRandom)
      val updatedWorld = world.copy(randomNumber = getNextRandom)
      worldDAO.updateRandom(updatedWorld)
      updatedWorld
    }
  }

  def getNextRandom: Int = {
    ThreadLocalRandom.current().nextInt(TestDatabaseRows) + 1
  }


  // Common code between Scala database code

  protected val TestDatabaseRows = 10000

  import models.WorldJsonHelpers.toJson

  def db = Action.async {
    getRandomWorlds(1).map { worlds =>
      Ok(Json.toJson(worlds.head))
    }
  }

  def queries(countString: String) = Action.async {
    val n = parseCount(countString)
    getRandomWorlds(n).map { worlds =>
      Ok(Json.toJson(worlds))
    }
  }

  def fortunes() = Action.async {
    getFortunes.map { dbFortunes =>
      val appendedFortunes = List(Fortune(0, "Additional fortune added at request time.")) ++ dbFortunes
      Ok(views.html.fortune(appendedFortunes))
    }
  }

  def update(queries: String) = Action.async {
    val n = parseCount(queries)
    updateWorlds(n).map { worlds =>
      Ok(Json.toJson(worlds))
    }
  }

  private def parseCount(s: String): Int = {
    try {
      val parsed = java.lang.Integer.parseInt(s, 10)
      parsed match {
        case i if i < 1 => 1
        case i if i > 500 => 500
        case i => i
      }
    } catch {
      case _: NumberFormatException => 1
    }
  }

}

File: frameworks/Java/play2-java/play2-java-ebean-hikaricp/app/controllers/Application.java
package controllers;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Random;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;
import java.util.concurrent.ThreadLocalRandom;

import javax.inject.Inject;

import models.Fortune;
import models.World;
import play.libs.Json;
import play.mvc.Controller;
import play.mvc.Result;
import utils.DatabaseExecutionContext;

public class Application extends Controller {

    private final DatabaseExecutionContext dbEc;

    @Inject
    public Application(final DatabaseExecutionContext dbEc) {
        this.dbEc = dbEc;
    }

    public CompletionStage<Result> db() {
        return getRandomWorlds(1).thenApply(worlds -> ok(Json.toJson(worlds.get(0))));
    }

    public CompletionStage<Result> queries(final String queries) {
        return getRandomWorlds(queryCount(queries)).thenApply(worlds -> ok(Json.toJson(worlds)));
    }

    public CompletionStage<Result> fortunes() {
        return CompletableFuture.supplyAsync(() -> {
            final List<Fortune> fortunes = Fortune.findAll();
            fortunes.add(new Fortune("Additional fortune added at request time."));
            Collections.sort(fortunes, (f1, f2) -> f1.message.compareTo(f2.message));

            return ok(views.html.fortunes.render(fortunes));
        }, dbEc);
    }

    public CompletionStage<Result> update(final String queries) {
        return getRandomWorlds(queryCount(queries)).thenApplyAsync(worlds -> {
            final Random random = ThreadLocalRandom.current();
            for (final World world : worlds) {
                world.randomNumber = (long) (random.nextInt(10000) + 1);
            }

            final List<World> updatedWorlds = World.save(worlds);
            return ok(Json.toJson(updatedWorlds));
        }, dbEc);
    }

    private int queryCount(final String queryCountString) {
        int queryCount;
        try {
            queryCount = Integer.parseInt(queryCountString, 10);
        } catch (final NumberFormatException e) {
            queryCount = 1;
        }
        if (queryCount < 1) {
            queryCount = 1;
        } else if (queryCount > 500) {
            queryCount = 500;
        }

        return queryCount;
    }

    private CompletionStage<List<World>> getRandomWorlds(final int n) {
        return CompletableFuture.supplyAsync(() -> {
            final Random random = ThreadLocalRandom.current();
            final List<World> worlds = new ArrayList<>(n);
            for (int i = 0; i < n; ++i) {
                long randomId = random.nextInt(10000) + 1;
                final World world = World.find(randomId);
                worlds.add(world);
            }
            return worlds;
        }, dbEc);
    }

}
